### ISSUE TITLE
[BUG] npm login/add-user --scope=@myScope do not set the scope on the .npmrc file

### ISSUE THREAD

**Issue by NachoAccedo** [npm/cli#3515]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

From https://docs.npmjs.com/cli/v7/commands/npm-adduser

Trying to login on a private artifactory with a defined scope **do not set** the scope on the ~/.npmrc file, i.e

`npm login --scope=@myScope --registry https://myArtifactory`

Generated ~/.npmrc
```
//myArtifactory/:_authToken=**
```

### Expected Behavior

For the npm `npm login --scope=@myScope --registry https://myArtifactory`
 command, we expect the next generated ~/.npmrc file
```
myScope=https://myArtifactory/
//myArtifactory/:_authToken=**
```

### Steps To Reproduce

npm login --scope=@myScope --registry https://myArtifactory

### Environment

- OS: OSX 11.4 Big Sur
- Node: v16.4.1
- npm: 7.18.1

**Comment by apastel** [comment]
Having the same issue as well and I seem to have isolated it down to beginning with `npm@7.7.0`, regardless of node version. The scope line is added when running npm up to 7.6.3, but then upgrading to 7.7.0 and attempting to login again with a clean ~/.npmrc, the scope line is not added.

**Comment by yuqu** [comment]
This is fixed in `npm@8.2.0`

**Comment by nlf** [comment]
closing as fixed
